tools/wpt/browser.py

```diff
@@ -1047,31 +1047,42 @@ def install(self, dest=None, channel=None):
         raise NotImplementedError
 
     def find_binary(self, venv_path=None, channel=None):
-        binary = None
+        if self.platform == "linux":
+            name = "microsoft-edge"
+            if channel == "stable":
+                name += "-stable"
+            elif channel == "beta":
+                name += "-beta"
+            elif channel == "dev":
+                name += "-dev"
+            # No Canary on Linux.
+            return find_executable(name)
+        if self.platform == "macos":
+            suffix = ""
+            if channel in ("beta", "dev", "canary"):
+                suffix = " " + channel.capitalize()
+            return "/Applications/Microsoft Edge%s.app/Contents/MacOS/Microsoft Edge%s" % (suffix, suffix)
```

Comment:
```suggestion
            return f"/Applications/Microsoft Edge{suffix}.app/Contents/MacOS/Microsoft Edge{suffix}"
```

If you like, it's slightly shorter and I recently learned about f-strings :)

Reply:
Thanks, updated to use f-strings throughout this class.